src/OrchardCore/OrchardCore/Shell/DefaultShellReleaseManager.cs

```diff
@@ -0,0 +1,42 @@
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Environment.Shell.Scope;
+
+namespace OrchardCore.Environment.Shell;
+
+public class DefaultShellReleaseManager : IShellReleaseManager
+{
+    private bool _release;
+    private bool _deferredTaskAdded;
+
+    public void SuspendReleaseRequest()
+    {
+        _release = false;
+    }
+
+    public void RequestRelease()
+    {
+        _release = true;
+
+        if (_deferredTaskAdded)
+        {
+            return;
+        }
+
+        _deferredTaskAdded = true;
+
+        ShellScope.AddDeferredTask(async scope =>
+        {
+            if (!_release)
```

Comment:
Does it work? The closure is not obvious here, it's ambiguous if this will be the value when the lambda is added or if it will evaluate the boolean once it is executed (what you want here).

A more robust option could be to set a flag in `ShellScope`, with ShellScope.Set<T>/Get<T>.

Reply:
I tried and it did work, though perhaps not in an exhaustive way.

Reply:
I was sure it works and you tried it, but reading the code I wondered how the value was copied. Which is why we usually have to copy it in closures to be sure it's not changed from the outside. I think the current implementation is fine, we could add a comment to disambiguate. I will definitely have the same question next time I see it.

Reply:
Indeed it works @sebastienros I also thought it is very weird to get that value with in the deferred call back.

Reply:
@sebastienros I made the changes you requested here https://github.com/OrchardCMS/OrchardCore/pull/15960